gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -73,6 +73,9 @@
   @BetaApi("The surface for customizing headers is not stable yet and may change in the future.")
   public abstract Map<String, String> getHeaders();
 
+  @BetaApi("The surface for customizing headers is not stable yet and may change in the future.")
+  protected abstract Map<String, String> getInternalHeaders();
```

Comment:
I might prefer "platform headers" or "library headers" as a name for this concept. I think I'd also like to rename the other headers to "custom headers" (or maybe "custom static headers").

Reply:
Keep as is as we discussed